src/videodownlinkbandwidthpolicy/VideoPriorityBasedPolicy.ts

```diff
@@ -31,6 +31,101 @@ class LinkMediaStats {
   rttMs: number = 0;
 }
 
+/** @internal */
+class PriorityBasedPolicyConfigManager {
+  private static readonly MINIMUM_DELAY = 2000; // keep the same as MIN_TIME_BETWEEN_SUBSCRIBE_MS
+  private static readonly MAXIMUM_DELAY = 8000;
+
+  private currentNetworkEvent: NetworkEvent;
+  private networkIncreaseTimestamp: number; // the last time network increases
+  private networkDecreaseTimestamp: number; // the last time network decreases
+  private networkEventChangeTimestamp: number; // the last time when network is changing from increase to decrease or decrease to increase
+
+  constructor(private videoPriorityBasedPolicyConfig: VideoPriorityBasedPolicyConfig) {
+    this.networkEventChangeTimestamp = 0;
+  }
+
+  getNetworkEventStartTimestamp(
+    numberOfParticipants: number,
+    prevEstimated: number,
+    currEstimated: number
+  ): [number, number] {
+    let networkEventStartTimestamp = 0;
+    let timeBeforeAllowSubscribeMs = 0;
+    const previousNetworkEvent = this.currentNetworkEvent;
+
+    if (currEstimated > prevEstimated) {
+      // if bw increases, we use recovery delay
+      this.currentNetworkEvent = NetworkEvent.Increase;
+      timeBeforeAllowSubscribeMs = this.getSubscribeDelay(
+        this.currentNetworkEvent,
+        numberOfParticipants
+      );
+
+      this.networkIncreaseTimestamp = Date.now();
+      // if bw is increasing, we want to look back to find when it starts to increase (the last moment bw is decreasing).
+      // e.g. [500, 400, 300, 400, 500, *600*], we are at 600. We know the bw is increasing since point 300, which is stored in networkDecreaseTimestamp.
+      networkEventStartTimestamp = this.networkDecreaseTimestamp;
+    } else if (currEstimated < prevEstimated) {
+      // if bw decreases, we use response delay
+      this.currentNetworkEvent = NetworkEvent.Decrease;
+      timeBeforeAllowSubscribeMs = this.getSubscribeDelay(
+        this.currentNetworkEvent,
+        numberOfParticipants
+      );
+
+      this.networkDecreaseTimestamp = Date.now();
+      // if bw is decreasing, we want to look back to find when it starts to decrease (the last moment bw is increasing) and select it as a window.
+      // see example above
+      networkEventStartTimestamp = this.networkIncreaseTimestamp;
+    } else {
+      // if bw is stable, we want to look back to find when it starts to become the current state
+      // e.g. [400, 500, 400, 400, 300, *300*], we are at the last 300. We know the bw is decreasing since point 500, which is stored in networkEventChangeTimestamp
+      networkEventStartTimestamp = this.networkEventChangeTimestamp;
+    }
+
+    if (
+      (previousNetworkEvent &&
+        previousNetworkEvent !== this.currentNetworkEvent) /* network state is changing */ ||
```

Comment:
optional (don't need to modify): I don't think some comments are unnecessary as the code is self-explaining well.